app/src/main/java/com/trevjonez/daggertest/main_activity/MainActivityComponent.java

```diff
@@ -16,18 +16,41 @@
 
 package com.trevjonez.daggertest.main_activity;
 
+import com.trevjonez.daggertest.fragment.MainActivityFragmentBinder;
 import com.trevjonez.inject.PlainComponent;
 import com.trevjonez.inject.activity.ActivityComponentBuilder;
 
+import javax.inject.Named;
+
+import dagger.Module;
+import dagger.Provides;
 import dagger.Subcomponent;
 
 /**
  * @author TrevJonez
  */
-@Subcomponent
+@Subcomponent (modules= {
+        MainActivityFragmentBinder.class,
+        MainActivityComponent.MainActivityModule.class
+})
 @MainActivityScope
 public interface MainActivityComponent extends PlainComponent<MainActivity> {
 
     @Subcomponent.Builder
-    interface Builder extends ActivityComponentBuilder<MainActivity, MainActivityComponent> { }
+    interface Builder extends ActivityComponentBuilder<MainActivity, MainActivityComponent> {
+        Builder activityModule(MainActivityModule module);
+    }
+
+    @Module
+    class MainActivityModule{
```

Comment:
Let's move this class to it's own file. I like to think of the component in terms of "Hey Dagger2 I need you to create this thing for me.", and the modules in terms of "I made this thing to help you do that stuff I asked for." Two separate and not necessarily related things.

And the biggest reason to do this is that some modules can be reused across different components for various reasons so separating it helps do that a bit more cleanly.